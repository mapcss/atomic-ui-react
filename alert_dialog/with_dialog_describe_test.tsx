import {
  anyString,
  assertSnapshot,
  describe,
  expect,
  fn,
  it,
  setupJSDOM,
} from "../dev_deps.ts";
import SSRProvider from "../ssr/ssr_provider.ts";
import WithDialogDescribe from "./with_dialog_describe.ts";
import AlertDialogProvider from "./alert_dialog_provider.ts";
import { render } from "@testing-library/react";

const describeTests = describe({
  name: "WithDialogDescribe",
  async beforeEach() {
    await setupJSDOM();
  },
});

it(describeTests, "should render as", (t) => {
  const { container } = render(
    <WithDialogDescribe>
      <div></div>
    </WithDialogDescribe>,
    {
      wrapper: ({ children }) => (
        <SSRProvider>
          <AlertDialogProvider>
            {children as never}
          </AlertDialogProvider>
        </SSRProvider>
      ),
    },
  );

  assertSnapshot(t, container.innerHTML);
});

it(describeTests, "should render children as props", (t) => {
  const mockFn = fn();
  render(
    <WithDialogDescribe>
      {(attributes) => {
        mockFn(attributes);
        return <div {...attributes}></div>;
      }}
    </WithDialogDescribe>,
    {
      wrapper: ({ children }) => (
        <SSRProvider>
          <AlertDialogProvider>
            {children as never}
          </AlertDialogProvider>
        </SSRProvider>
      ),
    },
  );

  expect(mockFn).toHaveBeenCalledWith({
    id: anyString(),
  });
});
