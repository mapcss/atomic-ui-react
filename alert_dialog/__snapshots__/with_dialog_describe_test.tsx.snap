export const snapshot = {};

snapshot[`WithDialogDescribe > should render as 1`] = `'<div role="alertdialog" aria-modal="true" aria-labelledby="atomic-ui-0-alert-dialog-title" aria-describedby="atomic-ui-0-alert-dialog-describe"><div id="atomic-ui-0-alert-dialog-describe"></div></div>'`;
